device/tt_device.h

```diff
@@ -801,7 +802,8 @@ class tt_SiliconDevice: public tt_device
     std::map<chip_id_t, std::unordered_map<std::int32_t, std::int32_t>> tlb_config_map = {};
     std::set<chip_id_t> all_target_mmio_devices;
     std::unordered_map<chip_id_t, std::vector<uint32_t>> host_channel_size;
-    std::function<std::int32_t(tt_xy_pair)> map_core_to_tlb;
+    std::unordered_map<chip_id_t, std::function<std::int32_t(tt_xy_pair)>> map_core_to_tlb_per_chip = {};
+    std::unordered_map<chip_id_t, bool> tlbs_init_per_chip = {};
```

Comment:
These are keyed by logical id, which can refer to a remote chip.  But the notion of inbound PCIe TLB has no meaning for a remote chip.

Reply:
Yes, there will be only local chip ids here. Do you mean I should make this more explicit?

Reply:
It's more a general remark.  This isn't the only place where a container looks like it contains per-chip data, but in reality only contains per-PCIe-chip data.  That is probably self-evident to people who have worked on the code for a while, but not so obvious to someone that doesn't have the context.

If you can think of something to clarify, feel free to put it in.

But I am pointing this situation mainly because it'd be nice to eliminate this pattern in the long run.

Reply:
Yes, I'm thinking about that. To my current understanding, only IO is on the critical perf path, and we do plan to provide "custom" solution for that.

Reply:
Added a comment for now. Maybe there will be a better way to explicitly state this in the future.